components/scream/src/share/io/scream_output_manager.cpp

```diff
@@ -0,0 +1,123 @@
+    new_output(out_params,true);
+
+    // Gather restart frequency info for restart history files
+    auto& freq_params = out_params.sublist("FREQUENCY");
+    restart_freq = freq_params.get<Int>("OUT_N");
+    restart_freq_units = freq_params.get<std::string>("OUT_OPTION");
+  }
+
+  // Construct and store an output stream instance for each output request.
+  // Typical output is controlled by parameter list which is stored in individual YAML files.
+  // A list of those files is passed to the output manager via the parameter list using the
+  // key "Output YAML Files".  See srd/share/io/tests for examples.
+  std::vector<std::string> empty;
+  auto& list_of_files = m_params.get("Output YAML Files",empty);
+  for (auto& it : list_of_files) {
+    ekat::ParameterList out_params(it);
+    parse_yaml_file(it,out_params);
+    out_params.set("CHECKPOINT FREQUENCY",restart_freq);
+    out_params.set("OUT_OPTION",restart_freq_units);
+    new_output(out_params,false);
+  }
+}
+/*===============================================================================================*/
+void OutputManager::run(util::TimeStamp& current_ts)
+{
+  for (auto& it : m_output_streams) {
+    it->run(current_ts);
+  }
+}
+/*===============================================================================================*/
+void OutputManager::finalize()
+{
+  for (auto& it : m_output_streams) {
+    it->finalize();
+  }
+  m_output_streams.clear();
+  m_io_comm = ekat::Comm();
+  m_params = ekat::ParameterList("");
+  m_field_mgr = nullptr;
+  m_runtype_restart = false;
+}
+/*===============================================================================================*/
+void OutputManager::
+new_output(const ekat::ParameterList& params, const bool model_restart_output)
+{
+  auto output = std::make_shared<output_type>(m_io_comm,params,m_field_mgr,
+                                              m_runtype_restart, model_restart_output);
+  m_output_streams.push_back(output);
+}
+/*===============================================================================================*/
+void OutputManager::make_restart_param_list(ekat::ParameterList& params)
+{
+  // Given the unique nature of restart files, this routine sets up the specific requirements.
+
+  //TODO change this to some sort of case_name, TODO: control so suffix is .r.nc
+  params.set<std::string>("FILENAME", "scorpio_restart_test");
+
+  // Parse the parameters that controls this output instance.
+  // Model restart themselves are instant snapshots (no averaging of any kind).
+  params.set<std::string>("AVERAGING TYPE","Instant");
+
+  if (not params.isParameter("GRID")) {
+    params.set<std::string>("GRID",m_field_mgr->get_grid()->name());
+  }
+  auto& freq_params = params.sublist("FREQUENCY");
+  freq_params.get<Int>("OUT_MAX_STEPS",1);
+
+  // Grab the restart fields from the field manager.
+  // If a developer wants a field to be stored in the restart file than they must add the "restart" group tag
```

Comment:
Is there a way we can use the DAG or the actions during the DAG to add this label by default to all "required" fields?  I have a hacky way to do that in a separate WIP PR, see https://github.com/E3SM-Project/scream/pull/1155/files#diff-ffc785b3f4b9ce9afd2510be2ae5df1eaa5f92d3b18d0a7ca67f0b7e70aea5a2R132
But I'm sure this can be done better.

Reply:
Yes, I wanted to get rid of this comment, and add code in the AD that deduces the restart group. I didn't get to that. If you want, I can do it here, otherwise we can do a separate PR (it's a very self-contained task, so I'd vote for PR, and assign to whomever got time).

Reply:
We can keep it in PR 1155 since it's one of the first places it will be used explicitly and then when it's ready to go you can review if there is a better way then what I did.